streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -885,8 +885,8 @@ private long pollPhase() {
             records = pollRequests(pollTime);
         } else if (state == State.PENDING_SHUTDOWN) {
             // we are only here because there's rebalance in progress,
-            // just poll with zero to complete it
-            records = pollRequests(Duration.ZERO);
+            // just long poll to give it enough time to complete it
+            records = pollRequests(pollTime);
```

Comment:
This is the main fix, see the PR description for full context. I was actually wondering if we shouldn't go even further and call `poll(MAX_VALUE)` instead, since there's really no reason to return from poll when the thread is shutting down but a rebalance is still in progress. Thoughts?

Reply:
I agree. I f there is nothing else to be done no point in timing it out

Reply:
LGTM. I think it was never correct to poll with zero in those cases, and just long polling even with MAX_VALUE is better.